internals/plan/plan.go

```diff
@@ -32,6 +32,32 @@ import (
 	"github.com/canonical/pebble/internals/osutil"
 )
 
+// LayerSectionExtension allows the plan layer schema to be extended without
+// adding centralised schema knowledge to the plan library.
+type LayerSectionExtension interface {
+	// ParseSection creates a new layer section containing the unmarshalled
+	// yaml.Node, and any additional section specifics it wishes to apply
+	// to the backing type. A nil LayerSection returned ensures that this
+	// section will be omitted by the caller.
```

Comment:
It's a bit strange to comment on the behavior of the caller in these functions.

I would expect the same idea to be formulated as "The return value is optional and a nil value signifies an absence of LayerSection".

Reply:
I've removed the caller wording. However, as the consumer of the interface, I do need to inform the implementer how to behave in order to achieve the expected results.

```https://pkg.go.dev/io#Reader```

For example, for ```io.Reader```:

```Callers should always process the n > 0 bytes returned before considering the error err. Doing so correctly handles I/O errors that happen after reading some bytes and also both of the allowed EOF behaviors.```